app/ui/SelectInput.tsx

```diff
@@ -66,33 +97,66 @@ export default function SelectInput({ className }: { className?: string }) {
   function handleOptionsChange(
     value: MultiValue<{ value: string; label: string }>,
   ) {
+    console.log(val);
     console.log(value);
 
-    const params = new URLSearchParams(searchParams);
+    // creating URLSearchParams object from search searchParams
+    const params = new URLSearchParams();
+    Object.keys(searchParams).forEach((key) => {
+      params.append(key, [searchParams[key]].flat().join(","));
+    });
+
+    // setting options param
     if (value.length > 0) {
       params.set("options", value.map((option) => option.value).join(","));
     } else {
       params.delete("options");
     }
-    router.replace(`${pathname}?${params.toString()}`);
+
+    // setting selected options state
+    setValue(value.map((option) => option.value));
+
+    // replacing current url with new params
+    router.replace(`/?${params.toString()}`);
+
+    // prefetching result page
+    router.prefetch(`/result?${params.toString()}`);
   }
 
   function handleRemDuplChange(value: boolean) {
     console.log(value);
 
-    const params = new URLSearchParams(searchParams);
+    // creating URLSearchParams object from search searchParams
+    const params = new URLSearchParams();
+    Object.keys(searchParams).forEach((key) => {
+      params.append(key, [searchParams[key]].flat().join(","));
+    });
+
+    // setting remDupl param
     params.set("remDupl", value.toString());
-    router.replace(`${pathname}?${params.toString()}`);
+
+    // setting remDupl state
+    setRemDupl(value);
+
+    // replacing current url with new params
+    router.replace(`/?${params.toString()}`);
+
+    // prefetching result page
+    router.prefetch(`/result?${params.toString()}`);
```

Comment:
Interesting, so when you select just 1 PARAM we would automatically prefetch results for that?

Reply:
Yes. But thanks to pointing out as now I see it prefetches even if there are no params selected, probably worth fixing